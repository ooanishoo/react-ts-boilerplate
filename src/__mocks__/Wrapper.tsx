import React from 'react'
import { BrowserRouter as Router } from 'react-router-dom'

/* a wrapper Router component that takes any React component and renders as it's child component */
type WrapperProps = {
    component?: React.ReactNode
}
export const Wrapper: React.FC<WrapperProps> = ({ component }: WrapperProps) => <Router>{component}</Router>
