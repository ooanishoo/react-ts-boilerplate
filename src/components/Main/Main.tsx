import React from 'react'
import Routes from '../../Routes'

const Main: React.FC = () => {
    return (
        <main data-testid="main" role="main">
            <Routes />
        </main>
    )
}

export default Main
