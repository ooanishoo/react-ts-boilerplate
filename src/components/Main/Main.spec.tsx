import { render } from '@testing-library/react'
import React from 'react'
import { Wrapper } from '../../__mocks__/Wrapper'
import Main from './Main'

describe('<Main/>', () => {
    test('should render without crashing', () => {
        const { container, getByTestId, queryByRole } = render(<Wrapper component={<Main />} />)
        expect(container).toMatchSnapshot()
        expect(getByTestId('main')).toBeInTheDocument()
        expect(queryByRole('main')).toBeTruthy()
    })
})
