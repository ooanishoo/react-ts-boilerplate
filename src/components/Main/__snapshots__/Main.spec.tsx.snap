// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main/> should render without crashing 1`] = `
<div>
  <main
    data-testid="main"
    role="main"
  >
    <div
      data-testid="home"
    >
      This is a Home Page
    </div>
  </main>
</div>
`;
